File: crates/remind-core/src/dto/note.rs
use crate::dto::block::BlockDTO;
use crate::entities::note::NoteIconType;
use uuid::Uuid;

#[derive(Clone, Debug)]
pub struct NoteDTO {
    pub id: Uuid,
    pub title: String,
    pub icon_type: NoteIconType,
    pub icon_data: String,
    pub workspace_id: Uuid,
    pub blocks: Vec<BlockDTO>,
    pub parent_note: Option<Uuid>,
}

#[derive(Clone, Debug)]
pub struct NoteCreateDTO {
    pub title: String,
    pub workspace_id: Uuid,
    pub parent_note: Option<Uuid>,
}

File: frontend/lib/api/types/note.ts
import { UUID } from "crypto";
import { Block } from "./block";

export interface Note {
  id: UUID;
  title: string;
  icon: {
    type: string;
    data: string;
  };
  workspace_id: UUID;
  blocks: Block[];
  parent: UUID | null;
}

export interface CreateNote {
  title: string;
  workspace_id: UUID | string;
  parent: UUID | string | null;
}

export interface UpdateNote {
  title: string | null;
}
